src/java.base/share/classes/java/lang/foreign/MemoryLayout.java

```diff
@@ -611,60 +687,14 @@ static PathElement sequenceElement() {
     String toString();
 
     /**
-     * Creates a padding layout with the given size.
+     * Creates a padding layout with the given bitSize and a bit-alignment of eight.
      *
-     * @param size the padding size in bits.
+     * @param bitSize the padding size in bits.
      * @return the new selector layout.
-     * @throws IllegalArgumentException if {@code size <= 0}.
+     * @throws IllegalArgumentException if {@code bitSize < 0} or {@code bitSize % 8 != 0}
```

Comment:
I'm not sure if this change in the `@throws` was deliberate - e.g. the new API seems to allow creation of padding layouts of zero size (which I did not realize). This can hide issues for code generators (e.g. I stumbled upon this when working on jextract, which was silently emitting zero-length paddings in some struct layouts). Perhaps better to revert to old semantics?

Reply:
Agreed. Having zero-length padding does not make much sense.